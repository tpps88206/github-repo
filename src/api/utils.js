export const getHeaders = () => ({
  Accept: 'application/vnd.github.v3+json',
  'Content-Type': 'Application/json',
});
