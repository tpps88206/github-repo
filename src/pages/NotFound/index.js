import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';

import Typography from '@material-ui/core/Typography';

import Page from '@/components/Page';
import PageContent from '@/components/Page/PageContent';

const NotFound = ({ redirect }) => {
  const history = useHistory();

  useEffect(() => {
    if (redirect) {
      history.replace(redirect);
    }
  }, []);

  return (
    <Page>
      <PageContent>
        <div className="mt-2">
          <Typography variant="h5">Page Not Found</Typography>
          {redirect && <Typography variant="body1">Redirecting...</Typography>}
        </div>
      </PageContent>
    </Page>
  );
};

export default NotFound;
