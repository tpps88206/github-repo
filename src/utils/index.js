import isEmpty from 'lodash/isEmpty';
import split from 'lodash/split';
import upperCase from 'lodash/upperCase';

export const generateAvatarCharacter = fullName => {
  if (!fullName) {
    return '';
  }
  const repoName = split(fullName, '/', 2);
  if (repoName.length > 1) {
    return upperCase(repoName[1].substring(0, 1));
  }
  return upperCase(repoName[0].substring(0, 1));
};

export const generateQueryParams = params => {
  let result = '';
  if (isEmpty(params)) {
    return result;
  }

  params.forEach((param, index) => {
    if (index > 0) {
      result = result + '&';
    }
    result = result + param.key + '=' + param.value;
  });
  return result;
};
