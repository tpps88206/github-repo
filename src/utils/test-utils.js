import React from 'react';
import { Provider } from 'react-redux';
// if you use Jest for testing and get the error with "use <Link> outside a <Router>"
// because just wrap your component with <BrowserRouter>
// see https://stackoverflow.com/questions/48640280/you-should-not-use-link-outside-a-router
import { BrowserRouter } from 'react-router-dom';

import { create } from 'jss';
import preset from 'jss-preset-default';

import { library } from '@fortawesome/fontawesome-svg-core';
import { fab } from '@fortawesome/free-brands-svg-icons';
import CssBaseline from '@material-ui/core/CssBaseline';
import { createMuiTheme, StylesProvider, ThemeProvider } from '@material-ui/core/styles';
import { render as rtlRender } from '@testing-library/react';

import theme from '@/constants/theme';
import configureStore from '@/redux/configureStore';
import styles from '@/styles';

library.add(fab);

const render = (ui, { ...renderOptions } = {}) => {
  const muiTheme = createMuiTheme(theme);
  const jss = create({
    plugins: [...preset().plugins],
  });
  const store = configureStore();

  jss.createStyleSheet(styles(muiTheme), { meta: 'global' }).attach();

  const Wrapper = ({ children }) => {
    return (
      <StylesProvider jss={jss}>
        <ThemeProvider theme={muiTheme}>
          <CssBaseline />
          <Provider store={store}>
            <BrowserRouter>{children}</BrowserRouter>
          </Provider>
        </ThemeProvider>
      </StylesProvider>
    );
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
