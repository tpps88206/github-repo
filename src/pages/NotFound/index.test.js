import React from 'react';

import { render } from '@/utils/test-utils';
import NotFound from './index';

describe('test Not Found', () => {
  it('renders NotFound without crashing', () => {
    const div = document.createElement('main');
    render(<NotFound />, div);
  });

  it('renders NotFound with snapshot', () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
  });
});
