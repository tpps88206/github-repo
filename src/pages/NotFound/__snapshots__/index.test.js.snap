// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Not Found renders NotFound with snapshot 1`] = `
<DocumentFragment>
  <main
    class="makeStyles-root-3 px-3 mt-9"
  >
    <section
      class="makeStyles-root-4 "
    >
      <div
        class="mt-2"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          Page Not Found
        </h5>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
